src/java.base/share/classes/java/time/format/DateTimeFormatterBuilder.java

```diff
@@ -165,7 +165,7 @@
      */
     private static final TemporalQuery<ZoneId> QUERY_REGION_ONLY = (temporal) -> {
         ZoneId zone = temporal.query(TemporalQueries.zoneId());
-        return (zone != null && zone instanceof ZoneOffset == false ? zone : null);
+        return (zone != null && (!(zone instanceof ZoneOffset)) ? zone : null);
```

Comment:
i find this code hard to read
`return (zone != null && !(zone instanceof ZoneOffset)) ? zone : null;`
seems better`

Reply:
Updated in 647bd6b as suggested by Michael Kuhlmann